elements/campaign-display/fields/campaign-request-field.js

```diff
@@ -19,7 +19,7 @@ const CampaignRequestField = {
 
     fetchCampaignSuccess(state, response, store) {
       state.requestInFlight = false;
-      store.actions.handleFetchComplete(response);
+      setImmediate(() => store.actions.handleFetchComplete(response));
```

Comment:
It would require a change to the store, but it wouldn't require any awkward queueing.